src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -2637,6 +2595,34 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
 
             break;
         }
+        case SMART_ACTION_COMPLETE_QUEST:
+        {
+            uint32 questId = e.action.quest.quest;
+            Quest const* quest = sObjectMgr->GetQuestTemplate(questId);
+            if (!quest)
+                break;
+
+            for (WorldObject* target : targets)
+            {
+                Player* player = Object::ToPlayer(target);
+                if (!player)
+                    continue;
+
+                QuestStatus questStatus = player->GetQuestStatus(questId);
+                if (questStatus == QUEST_STATUS_REWARDED)
```

Comment:
should we check completed status aswell here?

Reply:
we're already checking for status INCOMPLETE here https://github.com/TrinityCore/TrinityCore/pull/30048/files#diff-6d3483b74df32d157a6f8f227dacec1496d15fadfa8da3300e87fe5f6166e213R2617 and the tracking quests should only change status from NONE to REWARDED (COMPLETE doesn't make sense for those quests)